marbles/scripts/index.ts

```diff
@@ -0,0 +1,44 @@
+import { renderMarbleDiagram } from 'swirly-renderer-node';
+import { readdir, readFileSync, writeFileSync } from 'fs';
+import { join } from 'path';
+import { parseMarbleDiagramSpec } from 'swirly-parser';
+import { styles as defaultLight } from 'swirly-theme-default-light';
+import { DiagramStyles } from 'swirly-types';
+import * as SVGO from 'svgo';
+
+const diagramsPath = join(process.cwd(), 'marbles', 'diagrams');
+console.log(diagramsPath);
+readdir(diagramsPath, (err, files) => {
+    Promise.all(files.map(fileName => renderMarble(diagramsPath, fileName)))
+        .then(_ => console.log('All SVGs created'))
+        .catch(e => console.error('generating SVGs failed', e));
+});
+
+const renderMarble = (path: string, fileName: string): Promise<true> => {
+    const file = readFileSync(join(path, fileName));
+    const diagramSpec = parseMarbleDiagramSpec(file.toString());
+    const { xml: unoptXml, width, height } = renderMarbleDiagram(diagramSpec, { styles });
+    const optimizedSVGPromise = optimizeXml(unoptXml);
+    return optimizedSVGPromise.then((svgXML) => {
+        const svgFileName = fileName.split('.')[0] + '.svg';
+        const svgPath = join(process.cwd(), 'docs_app', 'assets', 'images', 'svgs', svgFileName);
+        writeFileSync(svgPath, svgXML, { encoding: 'utf-8', flag: 'w' });
+        return true;
```

Comment:
I don't think this does anything.

Reply:
yeah I prefered returning something instead of void, but the value is not really used, that's true